charmcraft/commands/store/client.py

```diff
@@ -43,12 +45,42 @@
 API_BASE_URL = 'https://api.staging.charmhub.io'
 STORAGE_BASE_URL = 'https://storage.staging.snapcraftcontent.com'
 
+TESTING_ENV_PREFIXES = ["TRAVIS", "AUTOPKGTEST_TMP"]
+
+
+def _get_os_platform(filepath=pathlib.Path("/etc/os-release")):
+    """Determine a system/release combo for an OS using /etc/os-release if available."""
+    system = platform.system()
+    release = platform.release()
+    machine = platform.machine()
+
+    if system == "Linux":
+        os_release = {}
+        try:
+            with filepath.open("r", encoding='utf-8') as f:
+                for line in f:
+                    if "=" in line:
```

Comment:
I left a check here as my test case `/etc/os-release` file was causing this to split and unpack an empty line which would panic. Rather than fixing the test case, I figured it is better safe than to cause a crash because someone/thing modified a file they shouldn't have.